plugins/database/hana/hana.go

```diff
@@ -148,148 +152,222 @@ func (h *HANA) CreateUser(ctx context.Context, statements dbplugin.Statements, u
 				"password":   password,
 				"expiration": expirationStr,
 			}
+
 			if err := dbtxn.ExecuteTxQuery(ctx, tx, m, query); err != nil {
-				return "", "", err
+				return newdbplugin.NewUserResponse{}, err
 			}
 		}
 	}
 
 	// Commit the transaction
 	if err := tx.Commit(); err != nil {
-		return "", "", err
+		return newdbplugin.NewUserResponse{}, err
+	}
+
+	resp := newdbplugin.NewUserResponse{
+		Username: username,
 	}
 
-	return username, password, nil
+	return resp, nil
 }
 
 // Renewing hana user just means altering user's valid until property
-func (h *HANA) RenewUser(ctx context.Context, statements dbplugin.Statements, username string, expiration time.Time) error {
-	statements = dbutil.StatementCompatibilityHelper(statements)
+func (h *HANA) UpdateUser(ctx context.Context, req newdbplugin.UpdateUserRequest) (newdbplugin.UpdateUserResponse, error) {
+	h.Lock()
+	defer h.Unlock()
+
+	// No change requested
+	if req.Password == nil && req.Expiration == nil {
+		return newdbplugin.UpdateUserResponse{}, nil
+	}
 
 	// Get connection
 	db, err := h.getConnection(ctx)
 	if err != nil {
-		return err
+		return newdbplugin.UpdateUserResponse{}, err
 	}
 
 	// Start a transaction
 	tx, err := db.BeginTx(ctx, nil)
 	if err != nil {
-		return err
+		return newdbplugin.UpdateUserResponse{}, err
 	}
 	defer tx.Rollback()
 
-	// If expiration is in the role SQL, HANA will deactivate the user when time is up,
-	// regardless of whether vault is alive to revoke lease
-	expirationStr, err := h.GenerateExpiration(expiration)
-	if err != nil {
-		return err
+	if req.Password != nil {
+		err = h.updateUserPassword(ctx, tx, req.Username, req.Password)
```

Comment:
Changing passwords and expirations will share a transaction. Is that intentional? I'm okay with it, I just want to make sure everyone else is good with it.

Reply:
I did, in fact, do that intentionally. The way I see it, if a user attempts to do two different actions and one fails, knowing the same request can be used is helpful. I'm not sure how others feel, but that's my preference with most interactions with an API.

Reply:
+1, if I update two pieces of configuration in the same API call to Vault, I would find it least surprising as a user if they either both succeed or both fail.

Reply:
Awesome! That's the same logic I had in my head, I just wanted to make sure everyone was on the same page.